jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/DataBatch.java

```diff
@@ -56,9 +59,15 @@ public DataBatch next() {
       try {
         int numRows = 0;
         int numElem = 0;
+        int numCol  = -1;
         List<LabeledPoint> batch = new ArrayList<>(batchSize);
         while (base.hasNext() && batch.size() < batchSize) {
           LabeledPoint labeledPoint = base.next();
+          if (numCol == -1) {
+            numCol = labeledPoint.size();
+          } else if (numCol != labeledPoint.size()) {
+            throw new RuntimeException("Feature size is not the same");
```

Comment:
any test to capture this exception?

Reply:
yes, it is in testCreateFromDataIteratorWithDiffFeatureSize from DMatrixTest.java

The exception cause DMatrix building failing,